src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -114,6 +114,75 @@ public static IResourceBuilder<PgAdminContainerResource> WithHostPort(this IReso
         });
     }
 
+    /// <summary>
+    /// Configures the host port that the pgweb resource is exposed on instead of using randomly assigned port.
+    /// </summary>
+    /// <param name="builder">The resource builder for pgweb.</param>
+    /// <param name="port">The port to bind on the host. If <see langword="null"/> is used random port will be assigned.</param>
+    /// <returns>The resource builder for pgweb.</returns>
+    public static IResourceBuilder<PgWebContainerResource> WithHostPort(this IResourceBuilder<PgWebContainerResource> builder, int? port)
+    {
+        return builder.WithEndpoint("http", endpoint =>
+        {
+            endpoint.Port = port;
+        });
+    }
+
+    /// <summary>
+    /// Adds an administration and development platform for PostgreSQL to the application model using pgweb.
+    /// </summary>
+    /// <param name="builder">The Postgres server resource builder.</param>
+    /// <param name="configureContainer">Configuration callback for pgweb container resource.</param>
+    /// <param name="containerName">The name of the container (Optional).</param>
+    /// <example>
+    /// Use in application host with a Postgres resource
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var postgres = builder.AddPostgres("postgres")
+    ///    .WithPgWeb();
+    /// var db = postgres.AddDatabase("db");
+    ///   
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(db);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </example>
+    /// <remarks>
+    /// This version the package defaults to the latest tag of the sosedoff/pgweb container image.
+    /// </remarks>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithPgWeb<T>(this IResourceBuilder<T> builder, Action<IResourceBuilder<PgWebContainerResource>>? configureContainer = null, string? containerName = null) where T : PostgresServerResource
+    {
+
+        if (builder.ApplicationBuilder.Resources.OfType<PgWebContainerResource>().SingleOrDefault() is { } existingPgWebResource)
+        {
+            var builderForExistingResource = builder.ApplicationBuilder.CreateResourceBuilder(existingPgWebResource);
+            configureContainer?.Invoke(builderForExistingResource);
+            return builder;
+        }
+        else
+        {
+            builder.ApplicationBuilder.Services.TryAddLifecycleHook<PgWebConfigWriterHook>();
+
+            containerName ??= $"{builder.Resource.Name}-pgweb";
+            var dir = Directory.CreateTempSubdirectory().FullName;
```

Comment:
do we have a hook or event when a resource stopped/finished?

I think we should add one API for that in `IDistributedApplicationLifecycleHook`

Reply:
This is actually a problem right now. @karolz-ms interested in your thoughts here. We sometimes drop temporary files in disk and I'd like to be able to try and ensure that they are cleaned up. We can probably have some kind of temporary file register that goes and does clean up but its highly likely that some of those files/directories are going to be locked because things like containers are spinning down.

This might be one of those things that DCP could help with?

Reply:
Yes, I think DCP would be a good place for logic that deals with "cleanup" tasks like removing temporary log files. 

There are probably many ways to do this, but the simplest one that will work today, is to put them inside the DCP session folder (which the app host runtime knows about). That folder is automatically deleted when DCP is shutting down. Just make sure the name is reasonably unique when creating the file.

Reply:
(we might be missing an API for the extensions that will tell them what the "temporary folder for the current application run" is)

Reply:
I think it would be good to be able to provide "cleanup paths" along with each resource when it shuts down. The path could either be a specific file or a whole directory.

Reply:
We could do that too. Feel free to open an issue and we can add it to our backlog for DCP.

Reply:
what is our approach for cleanup temp files for this PR?

Reply:
@mitchdenny - would this be a good usage of the new [Eventing API](https://github.com/dotnet/aspire/pull/5223)? We could have an event that fires when the resource is shut down.

Reply:
> what is our approach for cleanup temp files for this PR?

Since we already have the same issue with pgadmin:

https://github.com/dotnet/aspire/blob/4b4ebce8ef3f8db3fd43f0838d6734b46609f469/src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs#L101-L111

I'd be fine with continuing here to just create temp files and leave them, for now, until we get the clean up capability.

Reply:
Yes ... although it wouldn't be 100%. If someone kills the process via the debugger (most common scenario) I don't think we guarantee that this is run. But better than nothing I guess.